src/views/PasswordResetView.vue

```diff
@@ -0,0 +1,226 @@
+        <Button type="submit" id="reset-button" severity="danger">{{ $t('login.Reset') }}</Button>
+        <div class="backtologin" @click="backToLogin">{{ $t('login.Back to login') }}</div>
+      </Form>
+      <div v-else-if="passwordResetMode === 1" class="login-form">
+        <div class="sent-email">{{ $t('login.Email sent') }}</div>
+        <div class="backtologin" @click="backToLogin">{{ $t('login.Back to login') }}</div>
+      </div>
+      <Form v-else class="login-form" @submit="setNewPassword">
+        <span class="p-float-label with-error">
+          <InputText v-bind="password" id="password" size="large" name="password" type="password" :class="{'p-invalid': passwordForm.errors.value.password}"/>
+          <label :class="{'contains-text': password.modelValue }" for="password">{{ $t('login.New password') }}</label>
+        </span>
+        <small v-if="passwordForm.errors.value.password" class="p-error"><i class="pi pi-exclamation-circle"/>{{ passwordForm.errors.value.password }}</small>
+        <span class="p-float-label with-error">
+          <InputText v-bind="password2" id="password2" size="large" name="password" type="password" :class="{'p-invalid': passwordForm.errors.value.password2}"/>
+          <label :class="{'contains-text': password2.modelValue }" for="password2">{{ $t('login.Confirm password') }}</label>
+        </span>
+        <small v-if="passwordForm.errors.value.password2" class="p-error"><i class="pi pi-exclamation-circle"/>{{ passwordForm.errors.value.password2 }}</small>
+        <Button type="submit" id="reset-button" severity="danger">{{ $t('login.Reset') }}</Button>
+        <div class="backtologin" @click="backToLogin">{{ $t('login.Back to login') }}</div>
+      </Form>
+    </main>
+    <CopyrightBanner/>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { Form } from 'vee-validate';
+import CopyrightBanner from "@/components/CopyrightBanner.vue";
+import apiService from "@/services/ApiService";
+import router from "@/router";
+import { useForm } from "vee-validate";
+import InputText from "primevue/inputtext";
+import { onBeforeMount, ref } from "vue";
+import { useRoute } from "vue-router";
+import * as yup from "yup";
+import { toTypedSchema } from "@vee-validate/yup";
+
+const emailSchema = toTypedSchema(
+    yup.object({
+      email: yup.string().email().required(),
+    })
+);
+
+const passwordSchema = toTypedSchema(
+    yup.object({
+      password: yup.string().required(),
+      password2: yup.string().required(),
+    })
+);
+
+const emailForm = useForm({
+  validationSchema: emailSchema,
+});
+
+const passwordForm = useForm({
+  validationSchema: passwordSchema,
+});
+
+const passwordResetMode = ref(0);
+const email = emailForm.defineComponentBinds('email');
+const password = passwordForm.defineComponentBinds('password');
+const password2 = passwordForm.defineComponentBinds('password2');

```

Comment:
Might be nicer and more informative to refactor password2 to passwordConfirm?

Reply:
Updated in the latest commit to this branch.